src/main/java/org/checkerframework/specimin/AnnotationTargetRemoverVisitor.java

```diff
@@ -0,0 +1,242 @@
+
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.PackageDeclaration;
+import com.github.javaparser.ast.body.AnnotationDeclaration;
+import com.github.javaparser.ast.body.AnnotationMemberDeclaration;
+import com.github.javaparser.ast.body.ConstructorDeclaration;
+import com.github.javaparser.ast.body.EnumConstantDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.MethodDeclaration;
+import com.github.javaparser.ast.body.Parameter;
+import com.github.javaparser.ast.body.RecordDeclaration;
+import com.github.javaparser.ast.body.TypeDeclaration;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+import com.github.javaparser.ast.expr.ArrayInitializerExpr;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.expr.FieldAccessExpr;
+import com.github.javaparser.ast.expr.MarkerAnnotationExpr;
+import com.github.javaparser.ast.expr.NameExpr;
+import com.github.javaparser.ast.expr.NormalAnnotationExpr;
+import com.github.javaparser.ast.expr.SingleMemberAnnotationExpr;
+import com.github.javaparser.ast.expr.VariableDeclarationExpr;
+import com.github.javaparser.ast.type.TypeParameter;
+import com.github.javaparser.ast.visitor.ModifierVisitor;
+import com.github.javaparser.ast.visitor.Visitable;
+import java.lang.annotation.ElementType;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Removes all unnecessary ElementType values from each annotation declaration, based on their
+ * usages. Run this visitor after PrunerVisitor to ensure all unnecessary ElementTypes are removed.
+ */
+public class AnnotationTargetRemoverVisitor extends ModifierVisitor<Void> {
+  /** A Map of fully qualified annotation names to its ElementTypes. */
+  private final Map<String, Set<ElementType>> annotationToElementTypes = new HashMap<>();
+
+  /** A Map of fully qualified annotation names to their declarations. */
+  private final Map<String, AnnotationDeclaration> annotationToDeclaration = new HashMap<>();
+
+  /**
+   * Updates all annotation declaration {@code @Target} values to match their usages. Only removes
+   * {@code ElementType}s, doesn't add them. Call this method once after visiting all files.
+   */
+  public void removeExtraAnnotationTargets() {
+    for (Map.Entry<String, AnnotationDeclaration> pair : annotationToDeclaration.entrySet()) {
+      AnnotationExpr targetAnnotation = pair.getValue().getAnnotationByName("Target").orElse(null);
+
+      if (targetAnnotation == null) {
+        // This is most likely an existing definition. If there is no @Target annotation,
+        // we shouldn't add it
+        continue;
+      }
+
+      boolean useFullyQualified =
+          targetAnnotation.getNameAsString().equals("java.lang.annotation.Target");
```

Comment:
Since you've used the string `"java.lang.annotation.Target"` several times in this PR, it probably makes sense to pull it into a constant static field, just to avoid the possibility of typos, etc.